File: apis/DatabaseContextLibrary/Models/Adopter.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace DatabaseContextLibrary.models
{
    public class Adopter
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public string? UserName { get; set; }
        public string? PhoneNumber { get; set; }
        public string? Email { get; set; }
        public Guid AddressId { get; set; }
        public Address? Address { get; set; }
        public DateTime LastUpdateDate { get; set; }
        public AdopterStatus Status { get; set; }
    }

    public enum AdopterStatus
    {
        Active,
        Blocked,
        Deleted
    }
}

File: client/lib/adopter.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pet_share/address.dart';
import 'package:pet_share/shelter.dart';

part 'adopter.g.dart';

@JsonSerializable()
class Adopter extends User {
  Adopter(
      {required super.id,
      required super.userName,
      required super.phoneNumber,
      required super.email,
      required super.address,
      required this.status});

  AdopterStatusDTO status;

  @override
  Map<String, dynamic> toJson() => _$AdopterToJson(this);
  factory Adopter.fromJson(Map<String, dynamic> json) =>
      _$AdopterFromJson(json);
}

// ignore: constant_identifier_names
enum AdopterStatusDTO { Active, Blocked, Deleted }
